backend/src/integrations/sync.ts

```diff
@@ -1654,6 +1666,399 @@ const syncSecretsGitLab = async ({
+					nextPageToken = toString(currPageSecretsRes.nextPageToken);
+				} else {
+					nextPageToken = "";
+				}
+			} while (nextPageToken);
+
+			return secrets;
+		};
+
+		/**
+		 * Fetch latest version of the secret from gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.projectId
+		 * @param {string} param.secretId
+		 * @returns {GCPSecretManager.IAccessSecretVersionResponse}
+		 */
+		const getLatestSecretVersion = async ({
+			projectId,
+			secretId,
+		}: {
+			projectId: string;
+			secretId: string;
+		}): Promise<GCPSecretManager.IAccessSecretVersionResponse> => {
+			const url = `${INTEGRATION_GCP_SECRET_MANAGER_URL}/v1/projects/${projectId}/secrets/${secretId}/versions/latest:access`;
+
+			const latestSecretVersion: GCPSecretManager.IAccessSecretVersionResponse =
+				(await standardRequest.get(url, { headers: DEFAULT_HEADERS }))?.data;
+
+			return latestSecretVersion;
+		};
+
+		/**
+		 * Get secret versions list from gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.projectId
+		 * @param {string} param.secretId
+		 * @returns {GCPSecretManager.IListSecretVersionsResponse}
+		 */
+		const getSecretVersionsList = async ({
+			projectId,
+			secretId,
+		}: {
+			projectId: string;
+			secretId: string;
+		}): Promise<GCPSecretManager.IListSecretVersionsResponse> => {
+			const url = `${INTEGRATION_GCP_SECRET_MANAGER_URL}/v1/projects/${projectId}/secrets/${secretId}/versions`;
+
+			const secretVersionsList: GCPSecretManager.IListSecretVersionsResponse =
+				(await standardRequest.get(url, { headers: DEFAULT_HEADERS }))?.data;
+
+			return secretVersionsList;
+		};
+
+		/**
+		 * Create a secret in gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.projectId
+		 * @param {string} param.secretId
+		 * @returns {GCPSecretManager.Secret}
+		 */
+		const createSecret = async ({
+			projectId,
+			secretId,
+		}: {
+			projectId: string;
+			secretId: string;
+		}): Promise<GCPSecretManager.Secret> => {
+			const body = new GCPSecretManager.Secret({
+				name: `projects/${projectId}/secrets/${secretId}`,
+				replication: { automatic: {} },
+			});
+
+			const url = new URL(
+				`${INTEGRATION_GCP_SECRET_MANAGER_URL}/v1/projects/${projectId}/secrets`
+			);
+			url.searchParams.append("secretId", secretId);
+
+			const createdSecret: GCPSecretManager.Secret = (
+				await standardRequest.post(url.href, body, { headers: DEFAULT_HEADERS })
+			)?.data;
+
+			return createdSecret;
+		};
+
+		/**
+		 * Delete a secret in gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.projectId
+		 * @param {string} param.secretId
+		 * @returns {void}
+		 */
+		const deleteSecret = async ({
+			projectId,
+			secretId,
+		}: {
+			projectId: string;
+			secretId: string;
+		}) => {
+			const url = new URL(
+				`${INTEGRATION_GCP_SECRET_MANAGER_URL}/v1/projects/${projectId}/secrets/${secretId}`
```

Comment:
hmmm I think we should just destroy the latest secret version instead of deleting the entire secret so that it's less intrusive? As a user, I wouldn't want to unintentionally have my secret history be erased on GCP

Reply:
With this, you'll have to properly handle the fetching of the latest secret version though because afaik, GCP throws an error when you try to access a destroyed secret version

Reply:
Ok, I will add a check using `getLatestSecretVersion` function to make sure we check if it's already deleted.

Reply:
> hmmm I think we should just destroy the latest secret version instead of deleting the entire secret so that it's less intrusive? As a user, I wouldn't want to unintentionally have my secret history be erased on GCP

Yes, that's one case to avoid completely deleting the secret from GCP, but @dangtony98 can comment if infisical supports secret version yet, and if deleting a specific version is better.